src/tasks/Common/CompilerCache.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Build.Utilities;
+
```

Comment:
Since we are moving this to a new file, it's a great time to add `#nullable enable` :)